ambry-server/src/main/java/com.github.ambry.server/AmbryRequests.java

```diff
@@ -779,17 +779,40 @@ private AdminResponse handleBlobStoreControlRequest(DataInputStream requestStrea
     ServerErrorCode error;
     BlobStoreControlAdminRequest blobStoreControlAdminRequest =
         BlobStoreControlAdminRequest.readFrom(requestStream, adminRequest);
-    if (blobStoreControlAdminRequest.getNumReplicasCaughtUpPerPartition() > 0) {
-      PartitionId partitionId = blobStoreControlAdminRequest.getPartitionId();
-      if (partitionId != null) {
-        error = validateRequest(partitionId, RequestOrResponseType.AdminRequest);
-        if (error.equals(ServerErrorCode.No_Error)) {
-          if (blobStoreControlAdminRequest.shouldEnable()) {
-            // TODO: start BlobStore properly
-            error = ServerErrorCode.Temporarily_Disabled;
-          } else {
+    PartitionId partitionId = blobStoreControlAdminRequest.getPartitionId();
+    if (partitionId != null) {
+      error = validateRequest(partitionId, RequestOrResponseType.AdminRequest);
+      if (error.equals(ServerErrorCode.No_Error)) {
+        if (blobStoreControlAdminRequest.shouldEnable()) {
+          // start BlobStore properly
+          if (storageManager.startBlobStore(partitionId)) {
             Collection<PartitionId> partitionIds = Collections.singletonList(partitionId);
-            if (storageManager.disableCompactionForBlobStore(partitionId)) {
+            controlRequestForPartitions(RequestOrResponseType.GetRequest, partitionIds, true);
```

Comment:
I think we should modify this function to take in an `EnumSet`. That way we can compress the four lines into one and make this more concise (and some code in the stop part also)